train-controller/src/main/java/hu/bme/mit/train/controller/TrainControllerImpl.java

```diff
@@ -5,13 +5,30 @@
 import hu.bme.mit.train.interfaces.TrainController;
 
 import java.util.Date;
+import java.util.Timer;
+import java.util.TimerTask;
 
 public class TrainControllerImpl implements TrainController {
 
 	private int step = 0;
 	private int referenceSpeed = 0;
-	private int speedLimit = 100;
+	private int speedLimit = 99; //in order to make defense from Covid-19
 	public Table<String, String, Long> tachograph = HashBasedTable.create();
+	private Thread t; //i ve chosen thread as the descreption of task suggested
+	private int interval = 3000; //3sec periodic
+
+	public TrainControllerImpl() {
+		t = new Thread() {
+			public void run() {
+				t.run();
```

Comment:
Invoking the run command from here makes an infinite loop.

Also, you should start the thread, without it, this code does not do anything.

Also, this task is not recurring, change it to repeat the speed adjustments on an interval basis.